### PR TITLE
Publish layers

### PR DESCRIPTION
**Pull request by jwallace42**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | (add tickets here #1) |
| Primary OS tested on | (Ubuntu, MacOS, Windows) |
| Robotic platform tested on | (Steve's Robot, gazebo simulation of Tally, hardware turtlebot) |

---

## Description of documentation updates required from your changes

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_system_tests/src/error_codes/test_error_codes.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
 
 def main(argv=sys.argv[1:]):
     rclpy.init()
+    time.sleep(10)
```
Comment (SteveMacenski) [inline nav2_system_tests/src/error_codes/test_error_codes.py:29]: ? 10 seconds is alot to add to a test. What is this waiting for? Can this be reduced or based on an event?

Comment (jwallace42) [reply nav2_system_tests/src/error_codes/test_error_codes.py]: It is just waiting for the whole stack to come up. The same thing is done for the nav though poses test. 

https://github.com/ros-planning/navigation2/blob/main/nav2_system_tests/src/system/nav_through_poses_tester_node.py#L321

This should be resolved with the life cycle changes for python.

**Comment by BriceRenaudeau** [comment]
There is a crash due to this PR :boom: .

on cleanup, the layer_pub are reset but not removed from the layer_publishers vector. On activate, the activation of the layer_pub crashes.

Easy fix: add vector cleaning (`layer_publishers_.clear();`) after the following lines.

https://github.com/ros-planning/navigation2/blob/778c0faee41788c9e3323a420ff2c10122ba4aae/nav2_costmap_2d/src/costmap_2d_ros.cpp#L341-L343

**Comment by SteveMacenski** [comment]
https://github.com/ros-planning/navigation2/pull/3687 to resolve

**Pull request merged**